src/patterns/gen-ai/aws-qa-appsync-opensearch/index.ts

```diff
@@ -153,12 +153,33 @@ export interface QaAppsyncOpensearchProps {
    * and settings instead of the existing
    */
   readonly customDockerLambdaProps?: DockerLambdaCustomProps | undefined;
+
+  /**
+   * Optional. Allows to provide custom lambda code
+   * and settings instead of the existing
+   */
+  readonly sagemakerEndpointName?: string;
```

Comment:
I would remove all the logic related to a SG endpoint. If a user wants to use a SG endpoint, he can add the needed permissions to the lambda after the construct is created. The lambda can have a logic to automatically use an endpoint with the name matching a regex "idefics*" or something similar if that provider/model pair is passed through the mutation call, and send back an error message if the model is not supported (in this case, only idefics is supported)

Reply:
we can show the logic from this file in a sample

Reply:
done - 7d4d3a152fa0f0b4a28117814347191758cae246